### PR TITLE
svelte: Add search history mode to search input

### LINKED ISSUE #59208: Search input: Add history mode

### PR DESCRIPTION
**Pull request by fkling**
Closes #59208

This PR adds the search history mode to the search input. It consists of three parts:

- Refactoring some of the logic in the React implementation to share it with the Svelte implementation. That includes the logic around managing the search history.
- Adding the history button to the search input.
- Adding logic to add a successfull search to the history.

Note that I did not copty initialializing the history from event logs. This seemed to be a temporary solution to get some initial data when this feature was introduced (and should probably be removed from the React implementation).

https://github.com/sourcegraph/sourcegraph/assets/179026/8d60ac1a-f8b2-4e93-bdaf-429f17edd0c2

## Test plan

Manual testing and unit tests.

I'll try add an integration test later.

### CODE REVIEW

**Comment by fkling** [comment]
I realized that I ran into issues with my adjustments for loading rxjs. I need to look into fixing this. The code can still be reviewed but I need to spend some time to fix the build.

File: client/web-sveltekit/src/lib/search/input/recentSearches.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+interface RecentSearchesStore extends Readable<RecentSearch[] | null> {
+    addRecentSearch: (search: Omit<RecentSearch, 'timestamp'>) => void
+}
+
+/**
+ * Creates a store for recent searches. Must be called during component initialization.
+ */
+export function createRecentSearchesStore(): RecentSearchesStore {
+    const recentSearches = temporarySetting('search.input.recentSearches', [])
```
Comment (camdencheek) [inline client/web-sveltekit/src/lib/search/input/recentSearches.ts:17]: Do we use local storage in the current web app? Seems fine, just trying to figure out if I misunderstood how hsitory works in the react app.

File: client/web/src/search/input/recentSearches.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            }
+        }
+    }
+
+    /**
+     * Adds a new search to the top of the recent searches list.
+     * If the search is already in the recent searches list, it moves it to the top.
+     * If the list is full, the oldest search is removed.
+     */
+    private addOrMoveRecentSearchToTop(recentSearches: RecentSearch[], recentSearch: RecentSearch): RecentSearch[] {
```
Comment (camdencheek) [inline client/web/src/search/input/recentSearches.ts:66]: Should this use `options.persist`?

**Review by camdencheek — APPROVED** [review]
Nice! Looks great

File: client/web/src/search/input/recentSearches.ts (hunk 2 above)
Comment (fkling) [reply client/web/src/search/input/recentSearches.ts:66]: No, it returns the changed the list and the it's the caller responsibility to persist the changes if necessary.

**Pull request merged**